core/arch/arm/kernel/mutex_lockdep.c

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2018, Linaro Limited
+ */
+
+#include <assert.h>
+#include <kernel/thread.h>
+#include <kernel/lockdep.h>
+#include <sys/queue.h>
+#include <trace.h>
+
+#include "mutex_lockdep.h"
+
+/* Global graph of all mutexes used in the code */
+static struct lockdep_node_head graph = TAILQ_HEAD_INITIALIZER(graph);
+
+/* Protects @graph */
+static struct mutex m_graph = MUTEX_INITIALIZER;
```

Comment:
How about using a spinlock instead to protect the lockdep state?

Reply:
Good idea, will do.